File: RationMakerWebApiReactJS/ClientApp/src/components/Product.js
import React from "react";
import ModalDelete from "./ModalDelete";
import {Container} from "reactstrap";
import ApiAccessor from "./ApiAccessor";

const Product = ({ product, showCategories }) => {
    return (
            <tr key={product.id}>
                <td>{product.name}</td>
                <td>{product.gramms}</td>
                <td>{product.protein}</td>
                <td>{product.fats}</td>
                <td>{product.carbs}</td>
                <td>{product.calories}</td>
                {
                    showCategories ? <td>{product.category.name}</td> : <></>
                }
                <td>
                    <Container className="d-flex">
                        <ModalDelete product={product}/>
                        {/*<ApiAccessor product={product}/>*/}
                    </Container>
                </td>
            </tr>
    )
}
export default Product;

File: RationMakerWebApiReactJS/DataLayer/Models/Product.cs
namespace RationMakerWebApiReactJS.DataLayer.Models;

public class Product
{
    public int Id { get; set; }
    public string Name { get; set; }
    public double Gramms { get; set; }
    public double Protein { get; set; }
    public double Fats { get; set; }
    public double Carbs { get; set; }
    public double Calories { get; set; }

    public Category? Category { get; set; }
    public int CategoryId { get; set; }
}